internal/transform/operation.go

```diff
@@ -145,6 +157,71 @@ func mapOperationType(operation xdr.Operation) (string, error) {
+		operationTraceDescription = operationTrace.ManageBuyOfferResult.Code.String()
+	case xdr.OperationTypeManageSellOffer:
+		operationTraceDescription = operationTrace.ManageSellOfferResult.Code.String()
+	case xdr.OperationTypeCreatePassiveSellOffer:
+		operationTraceDescription = operationTrace.CreatePassiveSellOfferResult.Code.String()
+	case xdr.OperationTypeSetOptions:
+		operationTraceDescription = operationTrace.SetOptionsResult.Code.String()
+	case xdr.OperationTypeChangeTrust:
+		operationTraceDescription = operationTrace.ChangeTrustResult.Code.String()
+	case xdr.OperationTypeAllowTrust:
+		operationTraceDescription = operationTrace.AllowTrustResult.Code.String()
+	case xdr.OperationTypeAccountMerge:
+		operationTraceDescription = operationTrace.AccountMergeResult.Code.String()
+	case xdr.OperationTypeInflation:
+		operationTraceDescription = operationTrace.InflationResult.Code.String()
+	case xdr.OperationTypeManageData:
+		operationTraceDescription = operationTrace.ManageDataResult.Code.String()
+	case xdr.OperationTypeBumpSequence:
+		operationTraceDescription = operationTrace.BumpSeqResult.Code.String()
+	case xdr.OperationTypeCreateClaimableBalance:
+		operationTraceDescription = operationTrace.CreateClaimableBalanceResult.Code.String()
+	case xdr.OperationTypeClaimClaimableBalance:
+		operationTraceDescription = operationTrace.ClaimClaimableBalanceResult.Code.String()
+	case xdr.OperationTypeBeginSponsoringFutureReserves:
+		operationTraceDescription = operationTrace.BeginSponsoringFutureReservesResult.Code.String()
+	case xdr.OperationTypeEndSponsoringFutureReserves:
+		operationTraceDescription = operationTrace.EndSponsoringFutureReservesResult.Code.String()
+	case xdr.OperationTypeRevokeSponsorship:
+		operationTraceDescription = operationTrace.RevokeSponsorshipResult.Code.String()
+	case xdr.OperationTypeClawback:
+		operationTraceDescription = operationTrace.ClawbackResult.Code.String()
+	case xdr.OperationTypeClawbackClaimableBalance:
+		operationTraceDescription = operationTrace.ClawbackClaimableBalanceResult.Code.String()
+	case xdr.OperationTypeSetTrustLineFlags:
+		operationTraceDescription = operationTrace.SetTrustLineFlagsResult.Code.String()
+	case xdr.OperationTypeLiquidityPoolDeposit:
+		operationTraceDescription = operationTrace.LiquidityPoolDepositResult.Code.String()
+	case xdr.OperationTypeLiquidityPoolWithdraw:
+		operationTraceDescription = operationTrace.LiquidityPoolWithdrawResult.Code.String()
+	case xdr.OperationTypeInvokeHostFunction:
+		operationTraceDescription = operationTrace.InvokeHostFunctionResult.Code.String()
+	case xdr.OperationTypeExtendFootprintTtl:
+		operationTraceDescription = operationTrace.ExtendFootprintTtlResult.Code.String()
+	case xdr.OperationTypeRestoreFootprint:
+		operationTraceDescription = operationTrace.RestoreFootprintResult.Code.String()
+	default:
+		return operationTraceDescription, fmt.Errorf("Unknown operation type: %s", operationTrace.Type.String())
```

Comment:
I think `operationTrace.Type.String()` would cause an error if the OperationType is bad.

```
// String returns the name of `e`
func (e OperationType) String() string {
	name, _ := operationTypeMap[int32(e)]
	return name
}
```
And `operationTypeMap[int32(e)]` would return a invalid key in map or something

Reply:
I thought this default error was designed to raise when the Operation Type is _valid_, we just haven't mapped it in our case statements. The behavior mirrors the way we define [mapOperationType](https://github.com/stellar/stellar-etl/blob/master/internal/transform/operation.go#L83). My understanding is if the operationtype itself is bad, our code would already error and be unable to parse that operation.

Reply:
Do you want to change how we handle that in both functions? And if so, any guidance? Looks like the `operation.Type.String()` function does not return an error, so we don't have the ability to pass the error through

Reply:
> My understanding is if the operationtype itself is bad, our code would already error and be unable to parse that operation

Ah you're right. I think the only time this would actually be a problem is if XDR was updated ahead of what stellar-etl is using. And most likely something else would error out before it even reached this point

> Do you want to change how we handle that in both functions?

Nope it's fine as is